### PR TITLE
MM-401 Add indicators to the sidebar for offscreen unread channels

### PR DESCRIPTION
**Pull request by hmhealey**

### CODE REVIEW

File: web/react/components/sidebar.jsx [hunk 1]
```diff
@@ -132,11 +132,17 @@ module.exports = React.createClass({
...
+        this.updateUnreadIndicators();
+
+        $('.nav-pills__container').on('scroll', this.onScroll);
     },
     componentDidUpdate: function() {
         this.updateTitle();
+        this.updateUnreadIndicators();
     },
     componentWillUnmount: function() {
+        $('.nav-pills__container').off('scroll', this.onScroll);
```
Comment (ralder) [inline web/react/components/sidebar.jsx]: this may bind with react jsx attirbute onScroll

File: web/react/components/sidebar.jsx [hunk 2]
```diff
@@ -132,11 +132,17 @@ module.exports = React.createClass({
         $('.nav-pills__container').perfectScrollbar();
 
         this.updateTitle();
+        this.updateUnreadIndicators();
+
+        $('.nav-pills__container').on('scroll', this.onScroll);
```
Comment (ralder) [inline web/react/components/sidebar.jsx]: this may bind with react jsx attirbute onScroll

Comment (hmhealey) [inline web/react/components/sidebar.jsx]: That's a good point. I'll change that shortly.

**Comment by rgarmsen2295** [comment]
Looks good, +1

File: web/react/components/sidebar.jsx [hunk 3]
```diff
@@ -235,103 +247,143 @@ module.exports = React.createClass({
...
+    onScroll: function(e) {
+        this.updateUnreadIndicators();
+    },
+    onResize: function(e) {
+        this.updateUnreadIndicators();
+    },
+    updateUnreadIndicators: function() {
+        var container = $(this.refs.container.getDOMNode());
+
+        if (this.firstUnreadChannel) {
```
Comment (crspeller) [inline web/react/components/sidebar.jsx]: If the unread channels get updated so there is no first unread channel (perhaps from another device) How will the "$(this.refs.topUnreadIndicator.getDOMNode()).css('display', 'initial');" stuff get reset to not displaying?

Comment (hmhealey) [inline web/react/components/sidebar.jsx]: That's not a problem. When a channel gets changed to unread on another device, the component rerenders with both unread markers set back to display: none. updateUnreadIndicators is called afterwards to make the markers visible if they're still necessary.

Comment (crspeller) [inline web/react/components/sidebar.jsx]: Cool. I figured it was just a deficiency in my understanding.

**Comment by crspeller** [comment]
+1 From me.

**Comment by crspeller** [comment]
Waiting for @nickago

File: web/react/components/sidebar.jsx (hunk 3 above)
Comment (hmhealey) [inline web/react/components/sidebar.jsx]: No problem at all. I actually had to test it myself to make sure it worked as I thought it did.

**Comment by nickago** [comment]
Alright, +1, ready to merge

**Pull request merged**